client/client.go

```diff
@@ -465,6 +465,54 @@ func (c *client) parseRegionResponse(res *pdpb.GetRegionResponse) *Region {
 	return r
 }
 
+func (c *client) waitForRegion(ctx context.Context, key []byte) (*pdpb.GetRegionResponse, error) {
+	ctx, cancel := context.WithCancel(ctx)
+
+	ch := make(chan *pdpb.GetRegionResponse, len(c.connMu.clientConns))
+
```

Comment:
Maybe we can use `defer close(ch)` here rather than just close channel when received a non-nil `resp`?